lib/omniauth/shopify/scopes.rb

```diff
@@ -0,0 +1,48 @@
+module OmniAuth
+  module Shopify
+    class Scopes
+      SCOPE_DELIMITER = ','
+
+      def self.deserialize(scopes)
+        new(scopes.to_s.split(SCOPE_DELIMITER))
+      end
+
+      def initialize(scope_names)
+        @scopes = scope_names.map(&:strip).reject(&:empty?).to_set
+      end
+
+      def normalize
+        ignore_scopes = scopes.map { |scope| scope =~ /\A(unauthenticated_)?write_(.*)\z/ && "#{$1}read_#{$2}" }.compact
+
+        Scopes.new(scopes - ignore_scopes)
```

Comment:
Here you're explicitly writing out `Scopes.new` but above you only call `new` within the self.deserialize. Which way should we go with/which is the best practice? Do we only leave out the `Scopes` part if it's a class method?

Reply:
We can call `new` in `self.deserialize` because `new` is a class method and it will call `initialize` under the hood. When creating a new object inside of an instance, then you have to access the class to make a call to the class method `new`.